### PR TITLE
refactor(SigninRestoreWallet): code

### PR DESCRIPTION
**Pull request by Crazy-MIXER**
- Transferring the screen logic to components
- Move text to i18n
- Replace navigation hooks
- Change the import to absolute

### CODE REVIEW

File: src/components/singin-restore-wallet/singin-restore-wallet.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  const route = useTypedRoute<'restorePhrase'>();
+  const [seed, setSeed] = useState('');
+  const [disabled, setDisabled] = useState(false);
+
+  const checked = useMemo(
+    () => utils.isValidMnemonic(seed.trim()) || utils.isHexString(seed.trim()),
+    [seed],
+  );
+
+  const onDone = useCallback(() => {
```
Comment (gHashTag) [inline src/components/singin-restore-wallet/singin-restore-wallet.tsx:33]: Move method to screen level

**Review by gHashTag — CHANGES_REQUESTED** [review]
Fix comment

File: src/screens/signin-restore-wallet.tsx [hunk 2]
```diff
@@ -1,7 +1,20 @@
...
+import {utils} from 'ethers';
 
 import {SignInRestore} from '@app/components/singin-restore-wallet';
+import {useTypedNavigation, useTypedRoute} from '@app/hooks';
 
 export const SignInRestoreScreen = () => {
-  return <SignInRestore />;
+  const navigation = useTypedNavigation();
+  const route = useTypedRoute<'restorePhrase'>();
+  const [seed, setSeed] = useState('');
```
Comment (gHashTag) [inline src/screens/signin-restore-wallet.tsx:10]: Move this into component

File: src/screens/signin-restore-wallet.tsx [hunk 3]
```diff
@@ -1,7 +1,20 @@
...
+
+  const onDoneTry = () => {
```
Comment (gHashTag) [inline src/screens/signin-restore-wallet.tsx:12]: To use the state in a component
 ```js
const onDoneTry = (seed: string) => {
```

File: src/screens/signin-restore-wallet.tsx [hunk 4]
```diff
@@ -1,7 +1,20 @@
...
+import React, {useState} from 'react';
+import {utils} from 'ethers';
 
 import {SignInRestore} from '@app/components/singin-restore-wallet';
+import {useTypedNavigation, useTypedRoute} from '@app/hooks';
 
 export const SignInRestoreScreen = () => {
-  return <SignInRestore />;
+  const navigation = useTypedNavigation();
+  const route = useTypedRoute<'restorePhrase'>();
```
Comment (gHashTag) [inline src/screens/signin-restore-wallet.tsx:9]: use destructurization

**Review by gHashTag — CHANGES_REQUESTED** [review]
Fix comments

**Comment by Crazy-MIXER** [comment]
wait correction in src/i18n.ts to correctly resolve conflicts

File: src/screens/signin-restore-wallet.tsx (hunk 4 above)
Comment (gHashTag) [reply src/screens/signin-restore-wallet.tsx:10]: not done now

Comment (Crazy-MIXER) [reply src/screens/signin-restore-wallet.tsx]: Sorry, done now

**Review by gHashTag — APPROVED** [review]

**Review by vivalaakam — APPROVED** [review]

**Pull request merged**